cmd/frontend/graphqlbackend/graphqlutil/connection_resolver_test.go

```diff
@@ -0,0 +1,555 @@
+package graphqlutil
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/graph-gophers/graphql-go"
+
+	"github.com/sourcegraph/sourcegraph/internal/database"
+)
+
+var TOTAL_COUNT = int32(10)
+
+type TestConnectionNode struct {
+	id int
+}
+
+func (n TestConnectionNode) ID() graphql.ID {
+	return graphql.ID(fmt.Sprintf("%d", n.id))
+}
+
+type TestConnectionStore struct {
+	t                  *testing.T
+	TestComputeNodes   func(ctx context.Context, args *database.PaginationArgs)
+	ComputeTotalCalled int
+	ComputeNodesCalled int
+}
+
+func (s *TestConnectionStore) ComputeTotal(ctx context.Context) (*int32, error) {
+	s.ComputeTotalCalled = s.ComputeTotalCalled + 1
+	total := TOTAL_COUNT
+
+	return &total, nil
+}
+
+func (s *TestConnectionStore) ComputeNodes(ctx context.Context, args *database.PaginationArgs) ([]*TestConnectionNode, error) {
+	s.ComputeNodesCalled = s.ComputeNodesCalled + 1
+	nodes := []*TestConnectionNode{{id: 0}, {id: 1}}
+
+	s.TestComputeNodes(ctx, args)
```

Comment:
I would've assumed that it's this:

```suggestion
	nodes := s.TestComputeNodes(ctx, args)
```

Just as a nitpick/thought here.